lib/Doctrine/ORM/Query.php

```diff
@@ -306,12 +306,8 @@ protected function doExecute()
         $paramCount    = count($this->parameters);
         $mappingCount  = count($paramMappings);
 
-        if ($paramCount > $mappingCount) {
```

Comment:
This should remain?

Reply:
Setting more parameters than are mapped raises `QueryException::unknownParameter`:

https://github.com/doctrine/doctrine2/blob/695edb9fdcafbe81f3923317b1e9c5b4355be308/lib/Doctrine/ORM/Query.php#L396-L398